man/notify-selfcontained-example.c

```diff
@@ -0,0 +1,167 @@
+/* SPDX-License-Identifier: MIT-0 */
+
+/* Implement the systemd notify protocol without external dependencies.
+ * Supports both readiness notification on startup and on reloading,
+ * according to the protocol defined at:
+ * https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html
+ * This protocol is guaranteed to be stable as per:
+ * https://systemd.io/PORTABILITY_AND_STABILITY/ */
+
+#include <assert.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <time.h>
+#include <unistd.h>
+
+#define _cleanup_(f) __attribute__((cleanup(f)))
+
+bool reloading = 0;
+bool terminating = 0;
```

Comment:
please make this static. exporting these is icky

Reply:
you are supposed to use "volatile sig_atomic_t" for things like this btw.

See: https://www.gnu.org/software/libc/manual/html_node/Remembering-a-Signal.html